plugins/spectate/spectate.cpp

```diff
@@ -78,14 +77,41 @@ static PersistentDataItem pconfig;
 
 DFhackCExport command_result plugin_enable(color_ostream &out, bool enable);
 command_result spectate (color_ostream &out, std::vector <std::string> & parameters);
-#define COORDARGS(id) id.x, (id).y, id.z
+#define COORDARGS(id) id.x, id.y, id.z
+
+namespace DFHack {
+    namespace Units {
+        bool isInBox(df::unit* u,
+                     int16_t x1, int16_t y1, int16_t z1,
+                     int16_t x2, int16_t y2, int16_t z2) {
+            if (u->pos.x >= x1 && u->pos.x <= x2) {
+                if (u->pos.y >= y1 && u->pos.y <= y2) {
+                    if (u->pos.z >= z1 && u->pos.z <= z2) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+    }
```

Comment:
Could you move this to the *real* Units module and refactor `Units::getUnitsInBox()` to use it?